### PR TITLE
Refactor main JS architecture

### PR DESCRIPTION
**Pull request by pciavald**
- `npm run start:electron:fast` builds and runs without sourcemaps
- `npm run build:electron:fast` builds without sourcemaps
- GUI starts faster
- updating at runtime is disabled

### CODE REVIEW

File: src/app/core/rpc/rpx.service.ts [hunk 1]
```diff
@@ -1,19 +1,24 @@
 import { Injectable, NgZone } from '@angular/core';
 import { Observable } from 'rxjs/Observable';
 
+// multiwallet
+// import rxIpc from 'rx-ipc-electron/lib/renderer';
```
Comment (kewde) [inline src/app/core/rpc/rpx.service.ts]: Can't do this :P

Comment (pciavald) [reply src/app/core/rpc/rpx.service.ts]: i'm trying everything until I find something that at least works...

File: main.js [hunk 2]
```diff
@@ -25,34 +27,76 @@ let options;
...
 
-  daemon.init(_initWindow);
-  options = daemon.getOptions();
-}
+  options = _options.parse();
+  initMainWindow();
+
+  ipc.init();
+
+  daemon.check()
```
Comment (rynomster) [inline main.js]: This isn't supposed to be in main.js

File: main.js [hunk 3]
```diff
@@ -25,34 +27,76 @@ let options;
...
+    .then(()            => log.info('daemon already started'))
+    .catch(()           => daemonManager.init())
+    .then(()            => multiwallet.get())
+    .then(wallets       => ipc.promptWalletChoosing(wallets, mainWindow.webContents))
+    .then(chosenWallets => daemon.start(chosenWallets, () => log.info('daemon started')))
+});
+
+app.on('quit', function (event, exitCode) {
```
Comment (rynomster) [inline main.js]: This shouldn't be in main.js, we can create a new file to initialise everything

Comment (rynomster) [reply main.js]: Initialize and quit

File: modules/multiwallet/multiwallet.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+exports.get = function () {
+  return new Promise((resolve, reject) => {
+
+    spawn('ls', [ getWalletsPath() ]).then(files => {
+
+      files = files.stdout.split('\n');
+      // keep only wallet.dat and wallet_xxxx.dat files
+      files = files.filter(file => /(wallet\.dat|wallet_.+\.dat)/.test(file));
+      log.info('found wallets: ' + files);
```
Comment (rynomster) [inline modules/multiwallet/multiwallet.js]: Should be log.debug?

File: modules/rpc/cookie.js [hunk 5]
```diff
@@ -94,13 +94,14 @@ function mkDir(dirPath, root) {
...
 function getAuth(options) {
+
   if (options.rpcuser && options.rpcpassword) {
     return options.rpcuser + ':' + options.rpcpassword;
   }
 
-  // const COOKIE_FILE = findCookiePath() + `${options.testnet ? '/testnet' : ''}/.cookie`;
-  const COOKIE_FILE = findCookiePath() + (options.testnet ? '/testnet' : '') + '/.cookie';
   let auth;
+  let testnet = options.testnet ? '/testnet' : ''
```
Comment (rynomster) [inline modules/rpc/cookie.js]: Is an additional variable necessary here?

File: preload.js [hunk 6]
```diff
@@ -34,23 +35,26 @@ class SafeIpcRenderer {
...
+    this.removeListener     = protect(ipcRenderer.removeListener);
     this.removeAllListeners = protect(ipcRenderer.removeAllListeners);
-    this.send = protect(ipcRenderer.send);
-    this.sendSync = protect(ipcRenderer.sendSync);
-    this.sendToHost = protect(ipcRenderer.sendToHost);
   }
 }
 
 window.ipc = new SafeIpcRenderer([
+  'multiwallet',
```
Comment (rynomster) [inline preload.js]: We should prefix this with frontend or backend so we know where the channel is

File: preload.js [hunk 7]
```diff
@@ -34,23 +35,26 @@ class SafeIpcRenderer {
...
   'rpc-channel',
```
Comment (rynomster) [inline preload.js]: We should prefix this with frontend or backend so we know where the channel is registered when we call it

File: modules/rpc/cookie.js (hunk 5 above)
Comment (pciavald) [reply modules/rpc/cookie.js]: just for avoiding 80+ lines, should I remove it ?

Comment (rynomster) [reply modules/rpc/cookie.js]: changed the previous comment

File: src/app/core/rpc/rpx.service.ts [hunk 8]
```diff
@@ -23,7 +24,36 @@ export class RPXService {
...
+    window.ipc.on('rx-ipc-check-listener', (event, channel) => {
+      const replyChannel = 'rx-ipc-check-reply:' + channel;
+      if (this.listeners[channel]) {
+        event.sender.send(replyChannel, true);
+      } else {
+        event.sender.send(replyChannel, false);
+      }
+    });
+
+    window.ipc.on('front-choosewallet', (method, wallets) => {
```
Comment (kewde) [inline src/app/core/rpc/rpx.service.ts]: TODO: comment out or move to service.

Comment (kewde) [reply src/app/core/rpc/rpx.service.ts]: I'm seeing lots of commenting out of multiwallet code so I presume this must be as well

**Comment by rynomster** [comment]
Tests don't seem to pass

**Pull request closed without merging**